libwallet/assets/btc/dex-wallet.go

```diff
@@ -175,79 +328,522 @@ func (dw *DEXWallet) SignTx(tx *wire.MsgTx) error {
+	if err != nil {
+		return false, fmt.Errorf("GetCFilter error: %w", err)
+	}
+
+	if filter.N() == 0 {
+		return false, fmt.Errorf("unexpected empty filter for %s", blockHash)
+	}
+
+	var filterKey [gcs.KeySize]byte
+	copy(filterKey[:], blockHash[:gcs.KeySize])
+
+	matchFound, err := filter.MatchAny(filterKey, scripts)
+	if err != nil {
+		return false, fmt.Errorf("MatchAny error: %w", err)
+	}
+	return matchFound, nil
+}
+
+// blockIsMainchain will be true if the blockHash is that of a mainchain block.
+func (dw *DEXWallet) blockIsMainchain(blockHash *chainhash.Hash, blockHeight int32) bool {
+	if blockHeight < 0 {
+		var err error
+		blockHeight, err = dw.cl.GetBlockHeight(blockHash)
+		if err != nil {
+			log.Errorf("Error getting block height for hash %s", blockHash)
+			return false
+		}
+	}
+	checkHash, err := dw.cl.GetBlockHash(int64(blockHeight))
+	if err != nil {
+		log.Errorf("Error retrieving block hash for height %d", blockHeight)
+		return false
+	}
+
+	return *checkHash == *blockHash
+}
+
+// ownsInputs determines if we own the inputs of the tx.
+func (dw *DEXWallet) ownsInputs(txid string) bool {
+	txHash, err := chainhash.NewHashFromStr(txid)
+	if err != nil {
+		log.Warnf("Error decoding txid %q: %v", txid, err)
+		return false
+	}
+	txDetails, err := dw.walletTransaction(txHash)
+	if err != nil {
+		log.Warnf("walletTransaction(%v) error: %v", txid, err)
+		return false
+	}
+
+	for _, txIn := range txDetails.MsgTx.TxIn {
+		_, _, _, _, err = dw.w.FetchInputInfo(&txIn.PreviousOutPoint)
+		if err != nil {
+			if !errors.Is(err, wallet.ErrNotMine) {
+				log.Warnf("FetchInputInfo error: %v", err)
+			}
+			return false
+		}
+	}
+	return true
+}
+
+// outputSpendStatus will return the spend status of the output if it's found
+// in the TxDetails.Credits.
+func outputSpendStatus(details *wtxmgr.TxDetails, vout uint32) (spend, found bool) {
+	for _, credit := range details.Credits {
+		if credit.Index == vout {
+			return credit.Spent, true
+		}
 	}
-	return walletPeers, nil
+	return false, false
 }
 
-func (dw *DEXWallet) AddPeer(_ string) error {
-	return errors.New("AddPeer not implemented by DEX wallet")
+func confirms(txHeight, curHeight int32) int32 {
+	switch {
+	case txHeight == -1, txHeight > curHeight:
+		return 0
+	default:
+		return curHeight - txHeight + 1
+	}
 }
 
-func (dw *DEXWallet) RemovePeer(_ string) error {
-	return errors.New("RemovePeer not implemented by DEX wallet")
+func (dw *DEXWallet) syncedTo() waddrmgr.BlockStamp {
+	bs := dw.w.Manager.SyncedTo()
+	return waddrmgr.BlockStamp{
+		Height:    bs.Height,
+		Hash:      bs.Hash,
+		Timestamp: bs.Timestamp,
+	}
 }
 
-func (dw *DEXWallet) ListSinceBlock(start, end, syncHeight int32) ([]btcjson.ListTransactionsResult, error) {
-	return dw.w.ListSinceBlock(start, end, syncHeight)
+// accountInfo returns the account information of the wallet for use by the
+// exchange wallet.
+func (dw *DEXWallet) accountInfo() dexbtc.XCWalletAccount {
```

Comment:
Confirm this is still required. I think callers either want just the acct number or the acct name.

Reply:
updated